python3/src/stlib3/scene/contactheader.py

```diff
@@ -25,24 +25,22 @@ def ContactHeader(applyTo, alarmDistance, contactDistance, frictionCoef=0.0):
                 LocalMinDistance
             }
     '''
-    if applyTo.getObject("DefaultPipeline", warning=False) is None:
-            applyTo.addObject('DefaultPipeline')
+    rootNode.addObject('FreeMotionAnimationLoop')
+    rootNode.addObject('DefaultPipeline', verbose="0")
+    rootNode.addObject('BruteForceBroadPhase')
+    rootNode.addObject('BVHNarrowPhase')
 
-    applyTo.addObject('BruteForceDetection')
-
-    applyTo.addObject('RuleBasedContactManager', responseParams="mu="+str(frictionCoef),
+    rootNode.addObject('RuleBasedContactManager', responseParams="mu="+str(frictionCoef),
                                                     name='Response', response='FrictionContact')
-    applyTo.addObject('LocalMinDistance',
+    rootNode.addObject('LocalMinDistance',
                         alarmDistance=alarmDistance, contactDistance=contactDistance,
-                        angleCone=0.01)
-
-    if applyTo.getObject("FreeMotionAnimationLoop", warning=False) is None:
-            applyTo.addObject('FreeMotionAnimationLoop')
+                        angleCone=0.1)
+    rootNode.addObject('GenericConstraintSolver', tolerance=1e-5, maxIterations=5e2)
 
-    if applyTo.getObject("GenericConstraintSolver", warning=False) is None:
-            applyTo.addObject('GenericConstraintSolver', tolerance=1e-6, maxIterations=1000)
+    if rootNode.hasObject('SparseLDLSolver') is None:
+            rootNode.addObject('SparseLDLSolver')

```

Comment:
I think it still should be `GenericConstraintSolver` and not `SparseLDLSolver`, those are completely different solvers.